### ISSUE TITLE
Api hosted in service fabric can not pickup config

### ISSUE THREAD

**Issue by AndersMalmgren** [Microsoft/ApplicationInsights-aspnetcore#867]
If you are reporting bug/issue, please provide detailed Repro instructions.

## Repro Steps
Using a service fabric template project, I have made a repro project here https://github.com/AndersMalmgren/ServiceFabricLoggingRepro

Configure it for app insight doing 
```csharp
services
   .AddApplicationInsightsTelemetry()
```

##  Actual Behavior
It fails to pickup the config and log says 
Application Insights Telemetry (unconfigured)
If I explicit supply it with Configuration doing
```csharp
services
   .AddApplicationInsightsTelemetry(Configuration)
```

It works, but this is not how it behaves outside of service fabric.

## Expected Behavior
It should work the same way as if hosted outside of service fabric

## Version Info 
SDK Version : 2.7.0-beta3
.NET Version :	Core 2,2 						
How Application was onboarded with SDK(VisualStudio/StatusMonitor/Azure Extension) :
OS :
Hosting Info (IIS/Azure WebApps/ etc) : Service fabric with kestrel

**Comment by cijothomas** [comment]
services.AddApplicationInsightsTelemetry() is expected to pickup instrumentation key from appsettings.json. I don't see ikey configured in appseting.json https://github.com/AndersMalmgren/ServiceFabricLoggingRepro/blob/master/FabricApi/appsettings.json

The following is an eg: with instrumentation key.
{
      "ApplicationInsights": {
        "InstrumentationKey": "putinstrumentationkeyhere"
      },
      "Logging": {
        "LogLevel": {
          "Default": "Warning"
        }
      }
    }

**Comment by AndersMalmgren** [comment]
Hi @cijothomas 

Service fabric does not use that json. It uses a custom configuration provider

https://github.com/AndersMalmgren/ServiceFabricLoggingRepro/blob/master/FabricApi/FabricApi.cs#L45

It's populated using the service fabric settings file
https://github.com/AndersMalmgren/ServiceFabricLoggingRepro/blob/master/FabricApi/PackageRoot/Config/Settings.xml

To test it put a breakpoint in Startup.cs you will see that it populated correctly



I got this response from the service fabric team. I'm not sure its accurate though, your code correctly picks up the config in vanilla ASP Core.

> @AndersMalmgren , if you still want the same default behavior work for asp.net core and service fabric the same way, you need to open an issue for App Insight, asking them to change DefaultApplicationInsightsServiceConfigureOptions to use the Configuration from DI container rather than build their own ConfigurationBuilder.

**Comment by cijothomas** [comment]
@AndersMalmgren If SF don't use that json, AI SDK wont pickup settings.
See instrumentation key part from this doc:
https://docs.microsoft.com/en-us/azure/azure-monitor/app/asp-net-core-no-visualstudio#enabling-application-insights-server-side-telemetry

If you are configuring settings outside of the these, you need to call    .AddApplicationInsightsTelemetry(Configuration)

**Will keep this issue as an enhancement to modify SDK to pick config from DI.**

**Comment by AndersMalmgren** [comment]
This is false, from vanilla ASP Core I can use a custom configuration provider exactly the same way the service fabric configuration provider works and app insight will be able to pick it up without explicit supplying config instance.

**Comment by cijothomas** [comment]
https://github.com/Microsoft/ApplicationInsights- aspnetcore/blob/develop/src/Microsoft.ApplicationInsights.AspNetCore/Extensions/DefaultApplicationInsightsServiceConfigureOptions.cs#L31

We only read (by default) from appsettings.json and ENV.

**Comment by AndersMalmgren** [comment]
Yeah I was confusing it with native logging. Yeah, much better just injecting the IConfiguration

**Comment by chuanboz** [comment]
if injecting the IConfiguration, there is a priority issue that needs to be considered, that whether the injected IConfiguration has lower or higher priority than current default appSettings.json; It also needs to consider backwords compatibility of old behavior, otherwise it could be a breaking change.

**Comment by cijothomas** [comment]
@chuanboz Yes agreed.

At this stage, if not using the default appsettings.json, then there is workaround of supplying configuration using .AddApplicationInsightsTelemetry(Configuration) overload.
